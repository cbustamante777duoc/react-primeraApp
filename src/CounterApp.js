import React from 'react'
import PropTypes from 'prop-types';


//snippet para crear todo rapido es rafc
const CounterApp = ({ value }) => {

    //evento de mouse
    const handleAdd = (e)=>{
        console.log(e)
    }

    return (
        <>
            <h1>CounterApp</h1>

            <p>{value}</p>

            <button onClick={ handleAdd} >1+</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number

}


export default CounterApp;