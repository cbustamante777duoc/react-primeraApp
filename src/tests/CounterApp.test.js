import React from 'react';
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe('pruebas en el <CounterApp/> ', () => {

    let wrapper = shallow(<CounterApp/>);

    beforeEach(()=>{
        //el beforeEach cada ves que se ejecuta una prueba se ejecuta y eso hace que se 
        //receinice el componente con sus valores por defecto
        wrapper = shallow(<CounterApp/>);
    })
    
  

    test('debe mostrar los valores correctamente de <CounterApp/> en snapshot ', () => {

        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe de mostrar el valor por defecto de 100', () => {
        const wrapper = shallow(<CounterApp value={100}/>);

        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('100');

        
    })

    test('debe de incrementar con el boton de +1', () => {
        //toma el primer boton del CounterApp
        //const btnSuma = wrapper.find('button').at(0);
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('11');
        //console.log(btnSuma.html());    
    })

    test('debe de disminuir con el boton de -1', () => {
      
        
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('9');
        
    })

    test('debe poner el valor por defecto con el btn reset ', () => {
        const wrapper = shallow(<CounterApp value={110}/>);
        //primero se llama al boton de sumar para aumentar +1 = 111
        wrapper.find('button').at(0).simulate('click');
        //ahora se llama al boton de reiniciar = 110
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('110');
    })
    
    
    
    
    
})
