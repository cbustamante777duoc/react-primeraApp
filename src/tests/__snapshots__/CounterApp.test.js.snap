// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <CounterApp/>  debe mostrar los valores correctamente de <CounterApp/> en snapshot  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    1+
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
</Fragment>
`;
